### PR TITLE
log4j added

### PR DESCRIPTION
**Pull request by DmGrt**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -38,6 +38,11 @@
...
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
     </dependencies>
     <build>
         <plugins>
```
Comment (margarita-ignatieva) [inline pom.xml:48]: You also should add logs to .gitignore: logs/*

**Review by margarita-ignatieva — APPROVED** [review]

File: src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Root logger option
+log4j.rootLogger=INFO, stdout, file
+log4j.logger.org.hibernate=INFO
+log4j.logger.org.hibernate.SQL=DEBUG
+log4j.logger.org.hibernate.type.descriptor.sql=INFO
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]: are hibernate logs also recorded? do you want to do this?

File: src/main/java/com/dmgrt/cinema/Main.java [hunk 3]
```diff
@@ -15,8 +15,10 @@
 import com.dmgrt.cinema.service.UserService;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
+import org.apache.log4j.Logger;
 
 public class Main {
+    private static final Logger logger = Logger.getRootLogger();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dmgrt/cinema/Main.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(Main.class);
```

File: src/main/java/com/dmgrt/cinema/Main.java [hunk 4]
```diff
@@ -74,24 +76,24 @@ public static void main(String[] args) throws AuthenticationException {
         inceptionSession.setShowTime(LocalDateTime.now().plusMonths(5));
         movieSessionService.add(inceptionSession);
 
-        System.out.println(movieSessionService.findAvailableSessions(2L, LocalDate.now()));
-        System.out.println(movieSessionService.findAvailableSessions(3L, LocalDate.now()));
+        logger.info(movieSessionService.findAvailableSessions(2L, LocalDate.now()));
+        logger.info(movieSessionService.findAvailableSessions(3L, LocalDate.now()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dmgrt/cinema/Main.java:80]: not all souts in this file are moved to logs - please fix this

File: src/main/java/com/dmgrt/cinema/Main.java [hunk 5]
```diff
@@ -74,24 +76,24 @@ public static void main(String[] args) throws AuthenticationException {
...
 
         User jake = new User();
         jake.setPassword("tort");
         jake.setEmail("jake1956@meta.ua");
 
         jake = authenticationService.register(jake.getEmail(), jake.getPassword());
-        System.out.println("Registered: " + jake);
+        logger.info("Registered: " + jake);
 
         jake = authenticationService.login("jake1956@meta.ua", "tort");
```
Comment (KseniiaMakarova) [inline src/main/java/com/dmgrt/cinema/Main.java]: let's wrap this in try-catch and add log.warn with the informative message in `catch` block

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
**It will be great to have logging in DaoImpl classes as well**

File: src/main/java/com/dmgrt/cinema/Main.java [hunk 6]
```diff
@@ -76,17 +76,24 @@ public static void main(String[] args) throws AuthenticationException {
         inceptionSession.setShowTime(LocalDateTime.now().plusMonths(5));
         movieSessionService.add(inceptionSession);
 
-        logger.info(movieSessionService.findAvailableSessions(2L, LocalDate.now()));
-        logger.info(movieSessionService.findAvailableSessions(3L, LocalDate.now()));
+        System.out.println(movieSessionService.findAvailableSessions(2L, LocalDate.now()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dmgrt/cinema/Main.java]: there are still souts above(

File: src/main/java/com/dmgrt/cinema/Main.java [hunk 7]
```diff
@@ -76,17 +76,24 @@ public static void main(String[] args) throws AuthenticationException {
...
 
         User jake = new User();
         jake.setPassword("tort");
         jake.setEmail("jake1956@meta.ua");
 
-        jake = authenticationService.register(jake.getEmail(), jake.getPassword());
+        try {
+            jake = authenticationService.register(jake.getEmail(), jake.getPassword());
+        } catch (Exception e) {
+            logger.info("Can't register user." + e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dmgrt/cinema/Main.java]: can your `register()` method throw Exception?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
### **_please be attentive - not all comments from the previous review are fixed(_**

**Comment by KseniiaMakarova** [comment]
image

**Comment by DmGrt** [comment]
🤦‍♂️ I'm so sorry, I don't know what happened to me today

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**